app/process/watcher/message/watcher.go

```diff
@@ -129,5 +131,6 @@ func (cmw Watcher) processMessage(topicMsg mirror_node.Message, q *pair.Queue) {
 		return
 	}
 
-	q.Push(&pair.Message{Payload: msg})
+	// TODO: Figure this one out
+	q.Push(&queue.Message{Payload: msg, ChainId: big.NewInt(0)})
```

Comment:
Maybe we can include the source/target chain as part of the payload and read it from there?

Reply:
This is tricky, because the `Push` method actually is used in all the other logical flows, not only the ones related to what we are doing here. So if we parse the payload down the logical chain it might be problematic. That's why I added it separately so we can access it in the Queue and figure out which handler should take responsibility for the current incoming payload.

Reply:
When we add the EVM native assets transfer from Hedera to EVM, we will need to sign the target chain as part of the signature. This would mean that we will add the target chain in the payload. Lets leave this for the next PR when we add support for multiple EVM chains and address it there